Set dylib install_names.
Use .dylib suffix for dylib symlinks instead of .so.
Add support for DESTDIR.
--- makefile.osx.orig	2013-02-17 13:47:32.000000000 -0600
+++ makefile.osx	2018-11-05 09:18:55.000000000 -0600
@@ -163,8 +163,8 @@
 
 $(FASTDB_SHARED): $(OBJS)
 	rm -f $(FASTDB_SHARED)
-	$(CC) $(SHFLAGS) -o $(FASTDB_SHARED) $(OBJS) $(THRLIBS)
-	ln -f -s $(FASTDB_SHARED) libfastdb$(SUFF).so
+	$(CC) -install_name $(PREFIX)/lib/libfastdb_r.2.dylib $(SHFLAGS) -o $(FASTDB_SHARED) $(OBJS) $(THRLIBS)
+	ln -f -s $(FASTDB_SHARED) libfastdb$(SUFF).dylib
 
 $(CLI_LIB): $(CLI_OBJS)
 	rm -f $(CLI_LIB)
@@ -173,8 +173,8 @@
 
 $(CLI_SHARED): $(CLI_OBJS)
 	rm -f $(CLI_SHARED)
-	$(CC) $(SHFLAGS) -o  $(CLI_SHARED) $(CLI_OBJS) $(THRLIBS)
-	ln -f -s $(CLI_SHARED) libcli$(SUFF).so
+	$(CC) -install_name $(PREFIX)/lib/libcli_r.2.dylib $(SHFLAGS) -o  $(CLI_SHARED) $(CLI_OBJS) $(THRLIBS)
+	ln -f -s $(CLI_SHARED) libcli$(SUFF).dylib
 
 
 clitest.o: examples/clitest.c inc/cli.h
@@ -332,17 +332,17 @@
 
 
 install: subsql installlib
-	mkdir -p $(BINSPATH)
-	cp subsql $(BINSPATH)
-	strip $(BINSPATH)/subsql
+	mkdir -p $(DESTDIR)$(BINSPATH)
+	cp subsql $(DESTDIR)$(BINSPATH)
+	strip $(DESTDIR)$(BINSPATH)/subsql
 
 installlib: $(FASTDB_LIB) $(CLI_LIB) $(FASTDB_SHARED) $(CLI_SHARED)
-	mkdir -p $(INCSPATH)
-	cp $(INCS) $(INCSPATH)
-	mkdir -p $(LIBSPATH)
-	cp $(FASTDB_LIB) $(CLI_LIB) $(FASTDB_SHARED) $(CLI_SHARED) $(LIBSPATH)
-	cd $(LIBSPATH); ln -f -s $(FASTDB_SHARED) libfastdb$(SUFF).so
-	cd $(LIBSPATH); ln -f -s $(CLI_SHARED) libcli$(SUFF).so
+	mkdir -p $(DESTDIR)$(INCSPATH)
+	cp $(INCS) $(DESTDIR)$(INCSPATH)
+	mkdir -p $(DESTDIR)$(LIBSPATH)
+	cp $(FASTDB_LIB) $(CLI_LIB) $(FASTDB_SHARED) $(CLI_SHARED) $(DESTDIR)$(LIBSPATH)
+	ln -f -s $(FASTDB_SHARED) $(DESTDIR)$(LIBSPATH)/libfastdb$(SUFF).dylib
+	ln -f -s $(CLI_SHARED) $(DESTDIR)$(LIBSPATH)/libcli$(SUFF).dylib
 
 
 
